musl also supports endian.h, this makes systemc work on architectures other than x86.

diff -upr systemc-2.3.4.orig/src/sysc/packages/boost/detail/endian.hpp systemc-2.3.4/src/sysc/packages/boost/detail/endian.hpp
--- systemc-2.3.4.orig/src/sysc/packages/boost/detail/endian.hpp	2024-01-01 17:56:54.235418938 +0100
+++ systemc-2.3.4/src/sysc/packages/boost/detail/endian.hpp	2024-01-01 17:57:49.212244975 +0100
@@ -28,10 +28,10 @@
 #ifndef SC_BOOST_DETAIL_ENDIAN_HPP
 #define SC_BOOST_DETAIL_ENDIAN_HPP
 
-// GNU libc offers the helpful header <endian.h> which defines
+// GNU libc and musl libc offer the helpful header <endian.h> which defines
 // __BYTE_ORDER
 
-#if defined (__GLIBC__)
+#if defined (__linux__)
 # include <endian.h>
 # if (__BYTE_ORDER == __LITTLE_ENDIAN)
 #  define SC_BOOST_LITTLE_ENDIAN
